### ISSUE TITLE
Language is either not supported or WIP... c++mode

### ISSUE THREAD

**Issue by marlinstrub** [emacs-vs/ts-docstr#2]
This package looks very promising. When I try `ts-docstr-at-point` I get the error `Language is either not supported or WIP... c++mode`.

I load `tree-sitter` and `ts-docstr` as follows:

```elisp
(use-package tree-sitter
  :straight (tree-sitter :host github :repo "emacs-tree-sitter/elisp-tree-sitter"))

(use-package tree-sitter-langs
  :straight (tree-sitter-langs :host github :repo "emacs-tree-sitter/tree-sitter-langs"))

(use-package msgu
  :straight (msgu :host github :repo "jcs-elpa/msgu"))

(use-package ts-docstr 
  :straight (ts-docstr :host github :repo "emacs-vs/ts-docstr"))
```
I then visit a `cpp` file and call `M-x ts-docstr-at-point` on a function.

Am I doing this wrong?

**Comment by jcs090218** [comment]
Make sure you have `tree-sitter-mode` enabled in your buffer!

**Comment by marlinstrub** [comment]
Ah my bad, I had indeed misconfigured tree-sitter. I think it's setup correctly now, at least syntax highlighting works with it.

I'm not sure if I'm using this package correctly though. Let me try to explain what I'm doing. I created a test file with the following content:

![ss1](https://user-images.githubusercontent.com/11087545/200140226-31c8a725-0405-4b4a-8c7d-24ce5351da64.png)

If I insert a new line above line 1 and type `M-x ts-docstr-at-point` I get the following:

![ss2](https://user-images.githubusercontent.com/11087545/200140261-756220d0-31b4-42a7-af90-88d8315094be.png)

If I then place the cursor on line 7 (above `auto baz ...`) and type `M-x ts-docstr-at-point` again, nothing happens.

I noticed that the syntax highlighting is wrong after the first invocation of `ts-docstr-at-point`. I can fix it by temporarily removing the second star of the opening comment identifier (`/**`). If I do that and then place the cursor on line 7 and type `M-x ts-docstr-at-point` again, another comment is inserted:

![ss3](https://user-images.githubusercontent.com/11087545/200140486-86099b7d-765c-44cb-ab94-294302f46411.png)

Notice how the syntax highlighting of that second comment is wrong again.

I observe the same behaviour when I try with `emacs -Q` and load only the relevant packages. I'm on emacs version 29.0.50 (commit 8a5678906f).

Any pointers on how to fix this behaviour are much appreciated!

**Comment by jcs090218** [comment]
Yeah, I have the same experience and I suspect the issue is from the upstream `tree-sitter-langs`. I think if you edit buffer with or without this package you will experience something like that (any packages/operations that modified the buffer directly). Can you try and confirm the behavior?

Emacs devel is working on built-in tree-sitter, so I'm not quite sure the route of all `tree-sitter` packages (`ts-docstr`, `ts-fold`, `tree-sitter-langs`, etc). I think we will eventually replace all Elisp API with the built-in one. Then I guess this issue will then be resolved?

**Comment by marlinstrub** [comment]
Mhh. I don't use many packages that directly edit the buffer, but `yasnippet` works, for example. Did you have a particular package in mind that you wanted me to test?

I agree that it could well be a problem with upstream `tree-sitter`.

**Comment by nicodebo** [comment]
I think I have the same issue on python. After `M-x ts-docstr-at-point` the syntax highlighting seems to be broken in the whole buffer. After saving, the syntax highlighting is back to normal.

**Comment by jcs090218** [comment]
We might need to refresh the AST or font-lock highlighting after the insertion.

**Comment by ratnesh1729** [comment]
Thanks for the library. I face the same issue for python mode
